fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/properties/PrimaryKeyProperty.java

```diff
@@ -427,6 +428,12 @@ public Optional<List<Value>> visitSortPlan(@Nonnull final RecordQuerySortPlan so
             return primaryKeyFromSingleChild(sortPlan);
         }
 
+        @Nonnull
+        @Override
+        public Optional<List<Value>> visitRecursivePlan(@Nonnull final RecordQueryRecursivePlan recursivePlan) {
+            return commonPrimaryKeyFromChildren(recursivePlan);
```

Comment:
I think this is not correct. You probably need to do this for both the initial and the recursive part and see if they match, then propagate if they do.